src/global/project/environment.rs

```diff
@@ -0,0 +1,72 @@
+use std::{borrow::Borrow, fmt, str::FromStr};
+
+use miette::Diagnostic;
+use regex::Regex;
+use serde::{self, Deserialize, Deserializer};
+use thiserror::Error;
+
+/// Represents the name of an environment.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub(crate) struct EnvironmentName(String);
+
+impl EnvironmentName {
+    /// Returns the name of the environment.
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Borrow<str> for EnvironmentName {
+    fn borrow(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl fmt::Display for EnvironmentName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl PartialEq<str> for EnvironmentName {
+    fn eq(&self, other: &str) -> bool {
+        self.as_str() == other
+    }
+}
+
+impl<'de> Deserialize<'de> for EnvironmentName {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let name = String::deserialize(deserializer)?;
+        EnvironmentName::from_str(&name).map_err(serde::de::Error::custom)
+    }
+}
+
+/// Represents an error that occurs when parsing an environment name.
+///
+/// This error is returned when a string fails to be parsed as an environment name.
+#[derive(Debug, Clone, Error, Diagnostic, PartialEq)]
+#[error("Failed to parse environment name '{attempted_parse}', please use only lowercase letters, numbers and dashes")]
+pub struct ParseEnvironmentNameError {
```

Comment:
I think we can simplify it like this:

```
#[derive(Debug, Clone, Error, Diagnostic, PartialEq)]
#[error("Failed to parse environment name '{0}', please use only lowercase letters, numbers and dashes")]
pub struct ParseEnvironmentNameError(String);
```

based on `impl FromStr for PyPiPackageName` example

Reply:
It doesn't work. (and I would have felt that it is harder to read anyway^^)

Reply:
why it doesn't work?

Reply:
You are right, I didn't read careful enough.

I will ask the others for their opinion, since we have the same struct also in https://github.com/prefix-dev/pixi/blob/e3b2fd8dcb2ad4665b6298330fdee98834a5f051/crates/pixi_manifest/src/environment.rs#L91

Reply:
I talked with @baszalmstra. His argument was that `struct` is nicer here since one cannot guess immediately what `String` represents here. We could add a doc comment to clarify that, but I also think the `struct` makes more sense here.

Reply:
thanks!